### PR TITLE
Implement Codable for SpawnProcess

### PR DESCRIPTION
**Pull request by johari**

### CODE REVIEW

**Comment by johari** [comment]
We don't have CI on this repo. I ran tests locally and it succeeds.

File: Sources/llbuild2fx/SpawnProcess.swift [hunk 1]
```diff
@@ -365,3 +365,10 @@ public struct SpawnProcessResult: FXValue, FXTreeID {
         exitCode = codableValue
     }
 }
+
+extension SpawnProcessResult: Codable {
+    private enum CodingKeys: String, CodingKey {
+        case treeID
```
Comment (johari) [inline Sources/llbuild2fx/SpawnProcess.swift:371]: I noticed in `SpawnProcessResult` we have
```
    public var codableValue: Int32 {
        exitCode
    }
```
but the `Codable` conformance that I added in this PR also includes `treeID`.. is this desirable?

File: Sources/llbuild2fx/SpawnProcess.swift [hunk 2]
```diff
@@ -317,7 +317,7 @@ public struct ProcessInputTreeID: FXSingleDataIDValue, FXTreeID {
     }
 }
 
-public struct ProcessOutputTreeID: FXSingleDataIDValue, FXTreeID {
+public struct ProcessOutputTreeID: FXSingleDataIDValue, FXTreeID, Codable {
```
Comment (dmbryson) [inline Sources/llbuild2fx/SpawnProcess.swift:320]: Adding codable to an already existing FXValue doesn't sound right.  These are already being encoded and decoded from the CAS.

File: Sources/llbuild2fx/SpawnProcess.swift (hunk 1 above)
Comment (dmbryson) [reply Sources/llbuild2fx/SpawnProcess.swift:371]: Definitely not.   The treeID should not be ending up in the codableValue.   It must be stored in the refs.

**Review by dmbryson — CHANGES_REQUESTED** [review]

**Comment by johari** [comment]
Closing this PR for now. I will revisit the (mis)use of SpawnProcess in downstream dependencies instead.

**Pull request closed without merging**